code/modules/mob/living/simple_animal/hostile/retaliate/creacher/troll.dm

```diff
@@ -154,6 +161,93 @@
 	gender = PLURAL
 	icon_state = "Trolld"
 
+/mob/living/simple_animal/hostile/retaliate/troll/LoseTarget()
+	..()
+	if(!is_hidey)
+		return
+	if(health > 0)
+		icon_state = "Trollso"
+
+/mob/living/simple_animal/hostile/retaliate/troll/Moved()
+	. = ..()
+	if(!is_hidey)
+		return
+	if(!icon_state == "Troll")
+		icon_state = "Troll"
+
+
+/mob/living/simple_animal/hostile/retaliate/troll/GiveTarget()
+	..()
+	if(!is_hidey)
+		return
+	icon_state = "Trolla"
+
+/mob/living/simple_animal/hostile/retaliate/troll/after_creation()
+	..()
+	if(!is_hidey)
+		return
+	var/obj/item/organ/eyes/eyes = src.getorganslot(ORGAN_SLOT_EYES)
+	if(eyes)
+		eyes.Remove(src,1)
+		QDEL_NULL(eyes)
+	eyes = new /obj/item/organ/eyes/night_vision/nightmare
+	eyes.Insert(src)
+
+/mob/living/simple_animal/hostile/retaliate/troll/bog
+	name = "bog troll"
+	ai_controller = /datum/ai_controller/bog_troll
+	wander = FALSE		// bog trolls are ambush predators
+	turns_per_move = 4
+	is_hidey = TRUE
+
+	health = BOGTROLL_HEALTH
+	maxHealth = BOGTROLL_HEALTH
+	food_type = list(/obj/item/reagent_containers/food/snacks/meat,
+					/obj/item/bodypart,
+					/obj/item/organ)
+
+	base_intents = list(/datum/intent/simple/headbutt, /datum/intent/simple/bigbite)
+	melee_damage_lower = 30
+	melee_damage_upper = 50
+
+	TOTALCON = 16
+	TOTALSTR = 16
+	TOTALSPD = 3
+	TOTALEND = 15
+
+	defprob = 30
+	defdrain = 13
+
+/mob/living/simple_animal/hostile/retaliate/troll/cave
```

Comment:
please make this  harder to tame or something since it has the rock attack briars going to spam it